Sources/SwiftFrameCore/Workers/DecodableParser.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import Yams
+
+protocol KYDecoder {
+
+    func decode<T>(_ type: T.Type, from data: Data) throws -> T where T : Decodable
+
+}
+
+extension JSONDecoder: KYDecoder {}
+
+extension YAMLDecoder: KYDecoder {
+
+    func decode<T>(_ type: T.Type, from data: Data) throws -> T where T : Decodable {
+        guard let yamlString = String(data: data, encoding: .utf8) else {
+            throw NSError(description: "Could not read specified file")
+        }
+        return try decode(type, from: yamlString)
+    }
+
+}
+
+struct DecodableParser {
+
+    private enum FileFormat: String {
+        case json
+        case yaml
+
+        var decoder: KYDecoder {
+            switch self {
+            case .json:
+                return JSONDecoder()
+            case .yaml:
+                return YAMLDecoder()
+            }
+        }
+
+        init?(rawValue: String) {
+            switch rawValue {
+            case "json":
+                self = .json
+            case "yml", "yaml":
+                self = .yaml
+            default:
+                return nil
+            }
+        }
+    }
+
+    static func parseData<T>(fromURL url: URL) throws -> T where T: Decodable {
+        let data = try Data(contentsOf: url)
+        let decoder = try determineFileFormat(forURL: url).decoder
+
+        return try decoder.decode(T.self, from: data)
+    }
+
+    private static func determineFileFormat(forURL url: URL) throws -> FileFormat {
+        if let format = FileFormat(rawValue: url.pathExtension.lowercased()) {
```

Comment:
Could use a `guard` here or simply remove `else` since it's not needed due to the `return`.

Reply:
I'm not quite sure why the `else` branch is not needed? the init for `FileFormat` is failable so we'd fall-back to manually taking a look at the first line of the file if we can't already determine the format based on file extension

Reply:
What I meant is if you have

```
if foo {
    return bar
} else {
    foobar()
    return barfoo
}
```

the `else` is "superfluous" because due to the `return` you can just write it as


```
if foo {
    return bar
}
foobar()
return barfoo
```

Or you can also use a `guard` which requires a `return` in it's `else` branch so you can be certain that you didn't enter that branch when the code after `guard` is run.

Reply:
Ahh gotcha. You're right of course